### PR TITLE
Improve ZIO Mock documentation

### PR DESCRIPTION
**Pull request by ioleo**
Please review @psisoyev.

### CODE REVIEW

**Comment by ioleo** [comment]
Relates to #3268

File: docs/howto/howto_macros.md [hunk 1]
```diff
@@ -72,20 +72,28 @@ The `@mockable[A]` generates _capability tags_ and _mock layer_ into annotated o
...
+  object ProcessEvent extends Effect[AccountEvent, Nothing, Unit]
+  object RunCommand   extends Effect[Unit, Nothing, Unit]
 
-  private val envBuilder: URLayer[Has[Proxy], AccountObserver] =
-    ZLayer.fromService(invoke =>
-      new AccountObserver.Service {
-        def processEvent(event: AccountEvent) = invoke(ProcessEvent, event)
+  val compose: URLayer[Has[Proxy], AccountObserver] =
+    ZLayer.fromServiceM { proxy =>
+      withRuntime.map { rts =>
```
Comment (psisoyev) [inline docs/howto/howto_macros.md:91]: Not sure `withRuntime` is useful here

File: docs/howto/howto_macros.md [hunk 2]
```diff
@@ -72,20 +72,28 @@ The `@mockable[A]` generates _capability tags_ and _mock layer_ into annotated o
...
+        new AccountObserver.Service {
+          def processEvent(event: AccountEvent) = proxy(ProcessEvent, event)
+          def runCommand(): UIO[Unit]           = proxy(RunCommand)
```
Comment (psisoyev) [inline docs/howto/howto_macros.md:94]: Usually, empty brackets are added to methods without parameters that are doing side-effects. If we have `UIO` here there are no side-effects. At least there shouldn't be. However, I'm not sure what's the convention for effectful methods

File: docs/howto/mock_services.md [hunk 3]
```diff
@@ -279,9 +263,9 @@ To create expectations we use the previously defined _capability tags_:
 ```scala mdoc:silent
 import zio.test.Assertion._
 import zio.test.mock.Expectation._
-import zio.test.mock.MockSystem._
+import zio.test.mock.MockSystem
 
-val exp01 = Property(equalTo("foo"), value(Some("bar")))
+val exp01 = MockSystem.Property(equalTo("foo"), value(Some("bar")))
```
Comment (psisoyev) [inline docs/howto/mock_services.md:268]: Instead of `MockSystem` mock we could  `ExampleMock` here

File: docs/howto/howto_macros.md (hunk 1 above)
Comment (ioleo) [reply docs/howto/howto_macros.md:91]: The rts is always autogenerated by the macro.

**Comment by ioleo** [comment]
Rebased & resolved conflicts.

**Comment by ioleo** [comment]
@adamgfraser The failure is in FiberSpec, unrelated to this PR.

**Comment by adamgfraser** [comment]
@ioleo Will rerun.

**Comment by psisoyev** [comment]
sorry for being slow, this look great

**Pull request merged**